cmd/build/v2/config.go

```diff
@@ -114,16 +113,6 @@ func (oc oktetoBuilderConfig) GetAnonymizedRepo() string {
 	return oc.repository.GetAnonymizedRepo()
 }
 
-func (oc oktetoBuilderConfig) GetBuildContextHash(buildInfo *model.BuildInfo) string {
-	buildContext := buildInfo.Context
-	treeHash, err := oc.repository.GetTreeHash(buildInfo.Context)
-	if err != nil {
-		oktetoLog.Info("error trying to get tree hash for build context '%s': %w", buildContext, err)
-	}
-
-	return getBuildHashFromGitHash(buildInfo, treeHash, "tree_hash")
-}
-
 func (oc oktetoBuilderConfig) IsSmartBuildsEnabled() bool {
```

Comment:
How is `IsSmartBuildsEnabled` implemented? I think we have three different cases:
- Previous pull first approach based on commit -> default option
- Disable any kind of build optimization -> always build -> I think there is an envvar for this
- Smart builds based on build context 
Could we manage the three options with a single envvar?

Reply:
I would also have a single env var to control the behaviour, otherwise as a user I need to know and worry about 2 env vars and all combinations (true+true, true+false, etc)

Reply:
I prefer to have them separate. If the `OKTETO_SMART_BUILDS_ENABLED` is true then we don't check anything about the other variable. I agree that it can cause some troubles in the future but then I think the naming is not the best one.

If we go with only one variable, what should be the naming: `OKTETO_SMART_BUILD` with accepted values (`default`, `project` or `empty`/`none`/`buildcontext`)?

Reply:
@jLopezbarb which envvars names and values are currently implemented in this PR?

Reply:
@pchico83 Right now we have `OKTETO_SMART_BUILDS_USING_BUILD_CONTEXT` in order to use the build context instead the project commit and `OKTETO_SMART_BUILDS_ENABLED` to disable the feature.

Reply:
sounds good to me 👍